lambda_code/src/fhir_controller.py

```diff
@@ -99,23 +105,80 @@ def delete_immunization(self, aws_event):
             return self.create_response(500, unhandled_error.to_operation_outcome())
 
     def search_immunizations(self, aws_event) -> dict:
+        http_method = aws_event.get("httpMethod")
+        nhsNumberValue = None
+        diseaseTypeValue = None
+        nhsNumberParam = "-nhsNumber"
+        diseaseTypeParam = "-diseaseType"
+        if http_method == "POST":
+            # Check if the Content-Type is application/x-www-form-urlencoded
+            content_type = aws_event.get("headers", {}).get("Content-Type")
+            if content_type == "application/x-www-form-urlencoded":
+                # Parse the body of the request
+                body = aws_event["body"]
+                # Decode the base64-encoded body
+                decoded_body = base64.b64decode(body).decode("utf-8")
+                # Parse the decoded body as application/x-www-form-urlencoded
+                parsed_body = urllib.parse.parse_qs(decoded_body)
+                # Access individual form parameters and check if nhsNumber or diseaseType is present
+                nhsNumber_list = parsed_body.get(nhsNumberParam)
+                if nhsNumber_list:
+                    nhsNumberValue = nhsNumber_list[0]
```

Comment:
This silently ignores multiple values. It needs to be handle the same way as the query string.

Reply:
Can you please elaborate this